### Варіант № 3
### Завдання № 700(б)
Дана матриця $A$ порядку $n$. Отримати матрицю $AB$; елементи
матриці $B$ визначаються по формулі:
$b_{ij}=\begin{cases}
   \frac{1}{i+j-1}, &\text{якщо } i \leq j \\
   \frac{1}{i+j+1} &\text{в протилежному випадку }
\end{cases}$

In [5]:
from random import randint
from numpy import matrix

try:
    n = int(input("Enter the order of the matrix:"))
    if n > 0:
        a = matrix([[randint(-9, 9) for _ in range(n)] for _ in range(n)])
        b = matrix([[round(1 / (i + j - 1), 1) if i <= j else round(1 / i + j + 1, 1)
                     for j in range(1, n + 1)] for i in range(1, n + 1)])
        print(f"Matrix A:\n{a}\nMatrix B:\n{b}\nMatrix AB:\n{a * b}")
    else:
        print("Input error!")
except ValueError:
    print("Input error!")

Matrix A:
[[ 0  9  5  0  3  9 -9  7  6  9]
 [ 7 -3 -1 -6  4  9  5  3  4  9]
 [ 0  6 -1  1 -5 -9 -8  1  1  0]
 [-2  4  1  2  5  8 -6  0  1 -9]
 [ 6  4 -1 -2  7  5 -4  0  7 -7]
 [-4  2  2 -8 -6 -5  3  3  3 -6]
 [ 2 -8  3 -6  9  9 -3  0 -3 -5]
 [-6  7  0  1  4  8  6  7 -8  5]
 [ 1  2  6  1 -6  4  7 -7  3 -3]
 [ 3 -3  1  4 -1 -1 -1 -2 -4 -8]]
Matrix B:
[[ 1.   0.5  0.3  0.2  0.2  0.2  0.1  0.1  0.1  0.1]
 [ 2.5  0.3  0.2  0.2  0.2  0.1  0.1  0.1  0.1  0.1]
 [ 2.3  3.3  0.2  0.2  0.1  0.1  0.1  0.1  0.1  0.1]
 [ 2.2  3.2  4.2  0.1  0.1  0.1  0.1  0.1  0.1  0.1]
 [ 2.2  3.2  4.2  5.2  0.1  0.1  0.1  0.1  0.1  0.1]
 [ 2.2  3.2  4.2  5.2  6.2  0.1  0.1  0.1  0.1  0.1]
 [ 2.1  3.1  4.1  5.1  6.1  7.1  0.1  0.1  0.1  0.1]
 [ 2.1  3.1  4.1  5.1  6.1  7.1  8.1  0.1  0.1  0.1]
 [ 2.1  3.1  4.1  5.1  6.1  7.1  8.1  9.1  0.1  0.1]
 [ 2.1  3.1  4.1  5.1  6.1  7.1  8.1  9.1 10.1  0.1]]
Matrix AB:
[[  87.7   97.9  106.5  131.5  137.7   94.9  179.9  138.9   93.9    3.9]
 [  56.7   86.8  116.8  174.7  184

# Examples

### Example 1
Matrix A:
[[-9  5  4]
 [-1  1  3]
 [-1 -9  3]]
Matrix B:
[[1.  0.5 0.3]
 [2.5 0.3 0.2]
 [2.3 3.3 0.2]]
Matrix AB:
[[ 12.7  10.2  -0.9]
 [  8.4   9.7   0.5]
 [-16.6   6.7  -1.5]]
### Example 2
Matrix A:
[[-4 -1 -3 -8 -5]
 [-5  9  6 -4 -3]
 [ 6 -2  1  7  6]
 [ 4  3  4  5 -5]
 [ 5 -6 -5 -5  6]]
Matrix B:
[[1.  0.5 0.3 0.2 0.2]
 [2.5 0.3 0.2 0.2 0.2]
 [2.3 3.3 0.2 0.2 0.1]
 [2.2 3.2 4.2 0.1 0.1]
 [2.2 3.2 4.2 5.2 0.1]]
Matrix AB:
[[-42.  -53.8 -56.6 -28.4  -2.6]
 [ 15.9  -2.4 -27.9 -14.    0.7]
 [ 31.9  47.3  56.2  32.9   2.2]
 [ 20.7  16.1   2.6 -23.3   1.8]
 [-19.3 -12.6   3.5  29.5  -0.6]]
### Example 3
  Matrix A:
[[-5 -2 -1  2 -4  5 -7]
 [-4 -7  0 -7  7 -9  7]
 [-4  4 -9 -4  7  9  7]
 [ 4  4  8 -2  4  3  9]
 [-6  7  2 -4  7  3  2]
 [ 7  7 -9  7  9 -5  6]
 [ 9  4  6  7  7  4  7]]
Matrix B:
[[1.  0.5 0.3 0.2 0.2 0.2 0.1]
 [2.5 0.3 0.2 0.2 0.2 0.1 0.1]
 [2.3 3.3 0.2 0.2 0.1 0.1 0.1]
 [2.2 3.2 4.2 0.1 0.1 0.1 0.1]
 [2.2 3.2 4.2 5.2 0.1 0.1 0.1]
 [2.2 3.2 4.2 5.2 6.2 0.1 0.1]
 [2.1 3.1 4.1 5.1 6.1 7.1 0.1]]
Matrix AB:
[[-20.4 -18.5 -18.2 -31.9 -13.4 -50.7  -1.2]
 [-26.6 -11.2 -11.7  22.4 -15.3  47.3  -1.3]
 [ 26.4  29.6  76.9 116.7  97.9  49.6   1. ]
 [ 62.3  73.5  61.5  85.3  76.1  66.4   3. ]
 [ 33.5  31.1  33.4  62.4  31.5  14.5   1.1]
 [ 40.6  29.7  72.5  53.1   9.1  44.9   2.2]
 [ 87.1 104.8 109.   97.4  72.1  54.3   4.4]]